### PR TITLE
Testing Main Features

### LINKED ISSUE #59: Research .specs.ts methods
| User Story/Task        | P | SP |
|:------------------------|:-:|:--:|
| Research .specs.ts methods | 5 | 5 |

Right now, some components have default templates as tests, some tests are just all commented out. We want to have some automated tests going on that reflects the component /service being tested

------------------------------------------

------------------------------------------

## Acceptance Tests

* .specs.ts researched

------------------------------------------

## Related features/user stories

<!--~~~~HELPER~~~~~
Use bullet points like this:
* Related feature [#A]
* Related issue number [#B]
~~~~~~~~~~~~~~~-->

## Additional context
Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by tjosepo**
## Summary

This PR adds testing to all main features.

## Related Issue

* Closes #59

## Testing 

1. Checkout branch.
2. Run the command ```ng test``.

## Example

image

Look at all that satisfying green text.

### CODE REVIEW

**Review by CamilBouzidi — APPROVED** [review]
This is a very good PR. We struggled with testing a lot, so I'm glad that Jasmine turned out to work! Don't forget to update the docs (Unit Testing) if possible!

File: src/app/notifications/notifications.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        user: {
+          uid: 222,
+          displayName: 'Jack Black',
+          photoURL: 'https://lh3.googleusercontent.com/a-/AAuE7mCiim9TzdHHw8BmFQhNoKl879-GeH3qz6aLyO3mtA'
+        }
+      }])
+    };
+  }
+};
+
```
Comment (laurent-lao) [inline src/app/notifications/notifications.component.spec.ts:33]: Oh that's good

File: src/app/new-post/new-post-dialog/new-post-dialog.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const AngularFireFunctionsMock = {
+      httpsCallable: () => {
+        return () => of({});
+      }
+    };
+
+    const MatSnackBarMock = {
+      open: () => {}
+    };
+
```
Comment (laurent-lao) [inline src/app/new-post/new-post-dialog/new-post-dialog.component.spec.ts:51]: woah

**Review by laurent-lao — APPROVED** [review]
Nice

**Pull request merged**